src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -272,23 +275,43 @@
                                    :method method
                                    :snapshots (get methods [(:methodNamespace methodRepoMethod) (:methodName methodRepoMethod)])}
                                   (utils/restructure redacted? config methods editing? wdl-parse-error))]))
-        (create-section-header "Root Entity Type")
         (create-section
-         (if editing?
-           (if (seq entity-types)
-             (style/create-identity-select {:ref "rootentitytype"
-                                            :data-test-id "edit-method-config-root-entity-type-select"
-                                            :defaultValue rootEntityType
-                                            :style {:maxWidth 500}
-                                            :onChange #(swap! state assoc :autocomplete-list
-                                                              (build-autocomplete-list
-                                                               {:workspace-attributes workspace-attributes
-                                                                :entity-types entity-types
-                                                                :selected-entity-type (.. % -target -value)}))}
-               (mapv #(name (first %)) entity-types))
-             [:select {:style (assoc style/select-style :width 500) :disabled true}
-              [:option {} "No entities in workspace. Import some in the Data tab."]])
-           [:div {:style {:padding "0.5em 0 1em 0"}} rootEntityType]))
+         (style/create-detail-well
+          [:div {:style {:display "flex" :font-size "90%"}}
+           [:div {:style {:flex "1 1 60%"}}
+            [:div {} [Checkbox {:label [:span {:style {:font-weight 500 :color "#000"}}
+                                        "Configure inputs/outputs using the Workspace Data Model"
+                                        (dropdown/render-info-box {:text "TODO: Insert info and link here"})]
+                                :checked? entity-type?
+                                :disabled? (not (seq entity-type-options))
+                                :on-change (fn [new-value]
+                                             (if editing?
+                                               (swap! state assoc :entity-type? new-value)
+                                               (do (swap! locals assoc :toggle-entity-type? true)
```

Comment:
Can't actually change `entity-type?` here because of interactions with async calls sometimes made by `-begin-editing`. Instead we record that we want to toggle `entity-type?` and let `-begin-editing` actually toggle it at the right time.

Reply:
maybe add a comment here that mentions why we need the toggle-entity-type